ir/instr.cpp

```diff
@@ -1259,14 +1259,24 @@ StateValue ConversionOp::toSMT(State &s) const {
     fn = [](auto &&val, auto &to_type) -> StateValue {
       expr bv  = val.fp2sint(to_type.bits());
       expr fp2 = bv.sint2fp(val);
-      // -0.0 is converted to 0 and then to 0.0, though -0.0 is ok to convert
-      return { move(bv), val.isFPZero() || fp2 == val.roundtz() };
+      // -0.xx is converted to 0 and then to 0.0, though -0.xx is ok to convert
+      expr minus_one = expr::mkFloat(-1.0, val);
+      expr minus_zero = expr::mkFloat(-0.0, val);
+      return { move(bv),
```

Comment:
For both of these I think the more "pure" approach of:
```
expr valrtz = val.roundtz();
return { move(bv), valrtz.isFPZero() || fp2 == valrtz };
```
would work?

Reply:
It works, but I expect the SMT solver's performance is better for the range check because comparison is simpler than casts.
But this may not be true due to an unknown reason, so I'll leave Nuno to decide.

Reply:
The roudtz() value is already computed, so you'd better use it rather than creating 2 extra comparisons.